### PR TITLE
Do not change order of positive and negative values in PositionStack

### LINKED ISSUE #3390: geom_area with negative y not working in ggplot2 3.2.0
In `ggplot2 3.1.1` the area under the curve handles negative values
```r
library(ggplot2)

set.seed(115)

ggplot(
    data.frame(
        x = 1:100,
        y = cumsum(rnorm(n = 100, mean = 0, sd = 3))
    ),
    aes(x, y)) +
    geom_area(fill = "pink", alpha = 0.5) +
    geom_path()
```
image

In `ggplot2 3.2.0` something goes wrong:
```r
library(ggplot2)

set.seed(115)

ggplot(
    data.frame(
        x = 1:100,
        y = cumsum(rnorm(n = 100, mean = 0, sd = 3))
    ),
    aes(x, y)) +
    geom_area(fill = "pink", alpha = 0.5) +
    geom_path()
```
image

### PR DESCRIPTION
**Pull request by thomasp85**
fixes #3390 

@karawoo I have checked that this fix does not cause regressions in #1552. Basically what this does is avoid positive and negative y-values to get reordered, not fiddle with the group-based reordering

### CODE REVIEW

File: R/position-stack.r [hunk 1]
```diff
@@ -195,7 +195,7 @@ PositionStack <- ggproto("PositionStack", Position,
       )
     }
 
-    rbind(neg, pos)
+    rbind(neg, pos)[match(seq_len(nrow(data)), c(which(negative), which(!negative))),]
```
Comment (clauswilke) [inline R/position-stack.r]: I don't see immediately that this is always correct. Can there be problems with `NA`s? Or will `NA`s never reach this point?

Comment (thomasp85) [reply R/position-stack.r]: An `NA` value in ymax is pretty nonsensical and would also have broken the previous code. I can put in some guarding if you’d like nonetheless

Comment (clauswilke) [reply R/position-stack.r]: No strong opinion either way. Just make sure to look into why the CI builds fail.

Comment (thomasp85) [reply R/position-stack.r]: The failures are due to the downstream changes in sf, which have been fixed in master but not yet merged into the `v3.2.1-rc` branch

**Pull request merged**